plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/AppEngineLibrariesSelectorGroup.java

```diff
@@ -107,55 +104,209 @@ private static String getLibraryName(Library library) {
     }
   }
 
-  private void addDatabindingForDependencies() {
-    bindingContext = new DataBindingContext();
+  private void addDatabinding() {
+    bindingContext = new DataBindingContext(getDisplayRealm());
     for (Button libraryButton : libraryButtons) {
-      Library library = (Library) libraryButton.getData();
-      if (!library.getLibraryDependencies().isEmpty()) {
-        addDatabindingForDependencies(libraryButton);
-      }
+      addDatabindingForButton(libraryButton);
     }
   }
 
-  private void addDatabindingForDependencies(Button libraryButton) {
-    Library library = (Library) libraryButton.getData();
-    for (String libraryId : library.getLibraryDependencies()) {
-      Button dependencyButton = getButtonForLibraryId(libraryId);
-      if (dependencyButton != null) {
-        ISWTObservableValue libraryButtonSelection = WidgetProperties.selection().observe(libraryButton);
-        IObservableValue dependencyButtonSelection =
-            PojoProperties.value(Button.class, "selection").observe(getDisplayRealm(), dependencyButton);
-        IObservableValue dependencyButtonEnablement =
-            PojoProperties.value(Button.class, "enabled").observe(getDisplayRealm(), dependencyButton);
-
-        WritableValue intermediate = new WritableValue(false, Boolean.class);
-        bindingContext.bindValue(libraryButtonSelection, intermediate);
-        bindingContext.bindValue(dependencyButtonSelection, intermediate);
-        bindingContext.bindValue(dependencyButtonEnablement, intermediate,
-                                 new UpdateValueStrategy().setConverter(BooleanConverter.negate()),
-                                 new UpdateValueStrategy().setConverter(BooleanConverter.negate()));
-      }
+  private void addDatabindingForButton(final Button libraryButton) {
+    final Library library = (Library) libraryButton.getData();
+    ISWTObservableValue libraryButtonSelection = WidgetProperties.selection().observe(libraryButton);
+    ISWTObservableValue libraryButtonEnablement = WidgetProperties.enabled().observe(libraryButton);
+    // library selection UI -> model
```

Comment:
I think you need a comment here to say something along the lines of:

We have three bindings for each library:
  1. The first is a one-way binding of the checkbox selection state to add or remove the corresponding library from our selected-libraries list.
  2. The second is the opposite of the first, a one-way binding to set the checkbox selection state when the corresponding library has been selected or if it is a dependency of another selected library
  3. The last is a one-way binding to set the checkbox enablement when the corresponding library is a dependency of a selected library